lib/locale/format/number_format.dart

```diff
@@ -186,13 +186,13 @@ class NumberFormat {
         before = formatGroup(before);
       }
 
-      var length = prefix.length +
-              before.length +
-              after.length +
-              (zcomma ? 0 : negative.length),
-          padding = length < width
-              ? new List.filled((length = width - length + 1), '').join(fill)
-              : '';
+      int length = prefix.length +
+          before.length +
+          after.length +
+          (zcomma ? 0 : negative.length);
+      var padding = length < width
```

Comment:
I can change padding to a String, but strong mode has already correctly inferred it as a String. The reason I changed `length` to `int` is because if you just do `var` it is inferred as `num`, but later in this file you do `length >>= 1`, and `>>=` is only defined on `int`